File: mini-repos-IAC-v6/tasks/cdk-java/example/src/main/java/com/example/AppStack.java
package com.example;

import software.amazon.awscdk.CfnOutput;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.Tags;
import software.amazon.awscdk.services.s3.Bucket;
import software.constructs.Construct;

public class AppStack extends Stack {
    public AppStack(final Construct scope, final String id, final String namePrefix) {
        this(scope, id, namePrefix, null);
    }

    public AppStack(final Construct scope, final String id, final String namePrefix, final StackProps props) {
        super(scope, id, props);

        Bucket bucket = Bucket.Builder.create(this, "Main")
                .bucketName(namePrefix + "-bucket")
                .build();

        Tags.of(bucket).add("Name", namePrefix + "-bucket");
        Tags.of(bucket).add("Project", namePrefix);

        CfnOutput.Builder.create(this, "BucketName")
                .value(bucket.getBucketName())
                .description("S3 bucket name")
                .build();
    }
}

File: mini-repos-IAC-v9/tasks/cdk-ts/example/app-stack.ts
import { Stack, StackProps, CfnOutput, Tags } from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

interface AppStackProps extends StackProps {
  namePrefix: string;
}

export class AppStack extends Stack {
  constructor(scope: Construct, id: string, props: AppStackProps) {
    super(scope, id, props);

    const { namePrefix } = props;

    const bucket = new s3.Bucket(this, 'Main', {
      bucketName: `${namePrefix}-bucket`,
    });

    Tags.of(bucket).add('Name', `${namePrefix}-bucket`);
    Tags.of(bucket).add('Project', namePrefix);

    new CfnOutput(this, 'BucketName', {
      value: bucket.bucketName,
      description: 'S3 bucket name',
    });
  }
}
